### ISSUE TITLE
Nash testing infrastructure

### ISSUE THREAD

**Issue by tiago4orion** [NeowayLabs/nash#186]
- Builtin testing infrastructure?
- Go API for testing (like we did in [klb](https://github.com/NeowayLabs/klb/blob/master/tests/lib/nash/nash.go)

I'm inclined to add both the Go API *and* some nice way for people test theirs shell scripts without the need of using Go.

**Comment by katcipis** [comment]
I think both ideas are good. Go infra helps us test to catch deep nash bugs and the nash infrastructure would be great to test properly the scripts. I would just break in two completely separated issues for discussion and implementation.

**Comment by vitorarins** [comment]
I agree with @katcipis on treating things separately. Also I would give a higher priority to the nash test infrastructure.
